src/Cli/dotnet/CommandFactory/CommandSpec.cs

```diff
@@ -1,20 +1,18 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
-#nullable disable warnings
-
 using Microsoft.DotNet.Cli.CommandFactory.CommandResolution;
 
 namespace Microsoft.DotNet.Cli.CommandFactory;
 
 public class CommandSpec(
-    string? path,
+    string path,
     string? args,
-    Dictionary<string, string> environmentVariables = null)
+    Dictionary<string, string>? environmentVariables = null)
 {
     public string Path { get; } = path;
 
-    public string Args { get; } = args;
+    public string? Args { get; } = args;
```

Comment:
Having `Args` be potentially `null` seems suspicious, I considered changing this to the following but don't know enough about the code base to say if that is okay 

```csharp
public string Args { get; } = args ?? "";
```